benchexec/tracerx.py

```diff
@@ -0,0 +1,111 @@
+          Default target: x86_64-pc-linux-gnu
+          Host CPU: bdver1
+        """
+        stdout = self._version_from_tool(executable)
+        line = next(l for l in stdout.splitlines() if l.startswith("KLEE"))
+        line = line.replace("KLEE", "")
+        line = line.split("(")[0]
+        return line.strip()
+
+    def cmdline(self, executable, options, tasks, propertyfile=None, rlimits={}):
+        if propertyfile:
+            options += ["--property-file=" + propertyfile]
+        if benchexec.model.MEMLIMIT in rlimits:
+            options += ["--max-memory=" + str(rlimits[benchexec.model.MEMLIMIT])]
+        if benchexec.model.TIMELIMIT in rlimits:
+            options += ["--max-time=" + str(rlimits[benchexec.model.TIMELIMIT])]
+        if benchexec.model.WALLTIMELIMIT in rlimits:
+            options += ["--max-walltime=" + str(rlimits[benchexec.model.WALLTIMELIMIT])]
+        if benchexec.model.SOFTTIMELIMIT in rlimits:
+            options += [
+                "--max-cputime-soft=" + str(rlimits[benchexec.model.SOFTTIMELIMIT])
+            ]
+        if benchexec.model.HARDTIMELIMIT in rlimits:
+            options += [
+                "--max-cputime-hard=" + str(rlimits[benchexec.model.HARDTIMELIMIT])
+            ]
+
+        return [executable] + options + tasks
+
+    def name(self):
+        return "Tracer-X"
+
+    def determine_result(self, returncode, returnsignal, output, isTimeout):
+        """
+        Parse the output of the tool and extract the verification result.
+        This method always needs to be overridden.
+        If the tool gave a result, this method needs to return one of the
+        benchexec.result.RESULT_* strings.
+        Otherwise an arbitrary string can be returned that will be shown to the user
+        and should give some indication of the failure reason
+        (e.g., "CRASH", "OUT_OF_MEMORY", etc.).
+        """
+        for line in output:
+            if line.startswith("KLEE: ERROR: "):
+                if line.find("ASSERTION FAIL:") != -1:
+                    return result.RESULT_FALSE_REACH
+                elif line.find("memory error: out of bound pointer") != -1:
+                    return result.RESULT_FALSE_DEREF
+                elif line.find("overflow") != -1:
+                    return result.RESULT_FALSE_OVERFLOW
+                else:
+                    return "ERROR ({0})".format(returncode)
+            if line.startswith("KLEE: done"):
+                return result.RESULT_DONE
+        return result.RESULT_UNKNOWN
+
+    def get_value_from_output(self, lines, identifier):
+        # search for the text in output and get its value,
+        # stop after the first line, that contains the searched text
+        for line in lines:
+            if line.startswith("KLEE: done: ") and line.find(identifier + " = ") != -1:
+                startPosition = line.rfind("=") + 2
+                return line[startPosition:].strip()
```

Comment:
Code like this would crash if there is for example a truncated line in the output. I guess it would make sense to make it more robust, e.g., using `split` on `" = "`.

Reply:
fixed.